src/nvim/path.c

```diff
@@ -2080,8 +2080,14 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf, int len, int
   // expand it if forced or not an absolute path
   if (force || !path_is_absolute_path(fname)) {
     if ((p = vim_strrchr(fname, '/')) != NULL) {
-      STRNCPY(relative_directory, fname, p-fname);
-      relative_directory[p-fname] = NUL;
+      if (p == fname) {
+        // relative to root
```

Comment:
Ok. Only purpose of my suggestion was to improve the wording ("relative to root" is an odd way to say "absolute").  The unit test implies that `force=true` is required, so `path_is_absolute_path(fname)` must be true.